File: src/pucrs/domain/ProcessControlBlock.cs
using System;
using System.Collections.Generic;

namespace Modelos
{
    // é um bloco de informações de controle a respeito de um processo,
    // ele basicamente representa um processo;
    public class ProcessControlBlock
    {
        public Dictionary<string, int> Registradores { get; set; }

        public string ProcessID { get; }

        public State State { get; set; }

        public int Pc { get; set; }

        public int OffSet { get; set; }

        public int EnderecoLimite { get; set; }

        public int ParticaoAtual { get; set; }

        public ProcessControlBlock(string processID, int particaoAtual)
        {
            Registradores = new Dictionary<string, int>();

            Registradores.Add("r0", 0);
            Registradores.Add("r1", 0);
            Registradores.Add("r2", 0);
            Registradores.Add("r3", 0);
            Registradores.Add("r4", 0);
            Registradores.Add("r5", 0);
            Registradores.Add("r6", 0);
            Registradores.Add("r7", 0);

            this.ProcessID = processID;
            this.State = State.READY;
            this.ParticaoAtual = particaoAtual;
        }
    }

    public enum State
    {
        READY,
        RUNNING,
        WAITING,
        FINISHED,
        BLOCKED_BY_IO_OPERATION,
        ABORTED_DUE_TO_EXCEPTION
    }
}

File: src/pucrs/domain/ProcessControlBlock.java
package pucrs.domain;

import java.util.HashMap;
import java.util.Map;

public class ProcessControlBlock {

    private Map<String, Integer> registers;
    private String processID;
    private State state;
    private int pc = 0;
    private int offSet;
    private int limitAdress;
    private int actualPartition;

    public ProcessControlBlock(String processID, int actualPartition) {
        registers = new HashMap<>();

        registers.put("r0", 0);
        registers.put("r1", 0);
        registers.put("r2", 0);
        registers.put("r3", 0);
        registers.put("r4", 0);
        registers.put("r5", 0);
        registers.put("r6", 0);
        registers.put("r7", 0);

        this.processID = processID;
        this.state = State.READY;
        this.actualPartition = actualPartition;
    }

    public enum State {
        READY,
        RUNNING,
        WAITING,
        FINISHED,
        BLOCKED_BY_IO_OPERATION,
        ABORTED_DUE_TO_EXCEPTION
    }

    public Map<String, Integer> getRegisters() {
        return registers;
    }

    public void setRegisters(Map<String, Integer> registers) {
        this.registers = registers;
    }

    public String getProcessID() {
        return processID;
    }

    public void setProcessID(String processID) {
        this.processID = processID;
    }

    public State getState() {
        return state;
    }

    public void setState(State state) {
        this.state = state;
    }

    public int getPc() {
        return pc;
    }

    public void incrementPc() {
        pc++;
    }

    public void setPc(int pc) {
        this.pc = pc;
    }

    public int getOffSet() {
        return offSet;
    }

    public void setOffSet(int offSet) {
        this.offSet = offSet;
    }

    public int getLimitAdress() {
        return limitAdress;
    }

    public void setLimitAdress(int limitAdress) {
        this.limitAdress = limitAdress;
    }

    public int getActualPartition() {
        return actualPartition;
    }

}